4.1.1: pytest is failing

I'm trying to package your module as an rpm package. So I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

Just testted 4.2.1.
Only will quote summary
```console
============================================================================= warnings summary =============================================================================
../../../../../usr/lib/python3.8/site-packages/botocore/httpsession.py:41
  /usr/lib/python3.8/site-packages/botocore/httpsession.py:41: DeprecationWarning: 'urllib3.contrib.pyopenssl' module is deprecated and will be removed in a future release of urllib3 2.x. Read more in this issue: https://github.com/urllib3/urllib3/issues/2680
    from urllib3.contrib.pyopenssl import orig_util_SSLContext as SSLContext

../../../../../usr/lib/python3.8/site-packages/boto/plugin.py:40
  /usr/lib/python3.8/site-packages/boto/plugin.py:40: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

tests/integration/test_proxy.py:42
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_proxy.py:42: PytestDeprecationWarning: @pytest.yield_fixture is deprecated.
  Use @pytest.fixture instead; they are the same.
    @pytest.yield_fixture(scope="session")

tests/integration/test_tornado.py:53
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:53: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:65
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:65: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:77
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:77: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:90
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:90: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:103
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:103: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:118
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:118: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:141
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:141: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:156
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:156: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:168
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:168: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:187
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:187: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:212
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:212: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:224
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:224: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:242
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:242: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:258
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:258: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:274
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:274: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:288
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:288: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:295
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:295: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:309
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:309: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:322
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:322: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:337
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_tornado.py:337: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:53
  tests/integration/test_tornado.py:53: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_status_code will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:65
  tests/integration/test_tornado.py:65: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_headers will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:77
  tests/integration/test_tornado.py:77: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_body will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:90
  tests/integration/test_tornado.py:90: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_effective_url will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:103
  tests/integration/test_tornado.py:103: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:118
  tests/integration/test_tornado.py:118: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:141
  tests/integration/test_tornado.py:141: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_post will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:156
  tests/integration/test_tornado.py:156: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_redirects will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:168
  tests/integration/test_tornado.py:168: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:187
  tests/integration/test_tornado.py:187: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_gzip will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:212
  tests/integration/test_tornado.py:212: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:224
  tests/integration/test_tornado.py:224: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:241
  tests/integration/test_tornado.py:241: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:258
  tests/integration/test_tornado.py:258: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:273
  tests/integration/test_tornado.py:273: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:288
  tests/integration/test_tornado.py:288: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:295
  tests/integration/test_tornado.py:295: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:309
  tests/integration/test_tornado.py:309: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:322
  tests/integration/test_tornado.py:322: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:337
  tests/integration/test_tornado.py:337: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
    @pytest.mark.gen_test

tests/integration/test_wild.py:86
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.2.1/tests/integration/test_wild.py:86: PytestDeprecationWarning: @pytest.yield_fixture is deprecated.
  Use @pytest.fixture instead; they are the same.
    @pytest.yield_fixture(scope="session")

tests/integration/test_httpx.py: 22 warnings
  /usr/lib/python3.8/site-packages/rfc3986/_mixin.py:116: DeprecationWarning: Please use rfc3986.validators.Validator instead. This method will be eventually removed.
    warnings.warn(

tests/integration/test_httpx.py: 22 warnings
  /usr/lib/python3.8/site-packages/rfc3986/_mixin.py:172: DeprecationWarning: Please use rfc3986.validators.Validator instead. This method will be eventually removed.
    warnings.warn(

tests/integration/test_httpx.py: 22 warnings
  /usr/lib/python3.8/site-packages/rfc3986/_mixin.py:144: DeprecationWarning: Please use rfc3986.validators.Validator instead. This method will be eventually removed.
    warnings.warn(

tests/integration/test_httpx.py: 22 warnings
  /usr/lib/python3.8/site-packages/rfc3986/_mixin.py:191: DeprecationWarning: Please use rfc3986.validators.Validator instead. This method will be eventually removed.
    warnings.warn(

tests/integration/test_httpx.py: 22 warnings
  /usr/lib/python3.8/site-packages/rfc3986/_mixin.py:210: DeprecationWarning: Please use rfc3986.validators.Validator instead. This method will be eventually removed.
    warnings.warn(

tests/integration/test_httpx.py: 22 warnings
  /usr/lib/python3.8/site-packages/rfc3986/_mixin.py:229: DeprecationWarning: Please use rfc3986.validators.Validator instead. This method will be eventually removed.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [1] tests/integration/test_boto3.py:58: botocore version 1.27.62 does not use vendored requests anymore.
SKIPPED [1] tests/integration/test_boto3.py:76: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
SKIPPED [1] tests/integration/test_boto3.py:94: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
SKIPPED [1] tests/integration/test_requests.py:116: condition: sys.version_info >= (3, 6)
XFAIL tests/integration/test_tornado.py::test_status_code
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_status_code will be ignored
XFAIL tests/integration/test_tornado.py::test_headers
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_headers will be ignored
XFAIL tests/integration/test_tornado.py::test_body
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_body will be ignored
XFAIL tests/integration/test_tornado.py::test_effective_url
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_effective_url will be ignored
XFAIL tests/integration/test_tornado.py::test_auth
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth will be ignored
XFAIL tests/integration/test_tornado.py::test_auth_failed
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
XFAIL tests/integration/test_tornado.py::test_post
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_post will be ignored
XFAIL tests/integration/test_tornado.py::test_redirects
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_redirects will be ignored
XFAIL tests/integration/test_tornado.py::test_cross_scheme
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
XFAIL tests/integration/test_tornado.py::test_gzip
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_gzip will be ignored
XFAIL tests/integration/test_tornado.py::test_https_with_cert_validation_disabled
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raises_in_future
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raise_error_disabled
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raises_in_future
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raise_error_disabled
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_with_decorator_use_cassette
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_exception_can_be_caught
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_references_get_patched
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_instances_get_patched
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_request_time_is_set
  reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
FAILED tests/integration/test_boto.py::test_boto_without_vcr - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAu...
FAILED tests/integration/test_boto.py::test_boto_medium_difficulty - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['...
FAILED tests/integration/test_boto.py::test_boto_hardcore_mode - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['Hmac...
FAILED tests/integration/test_boto.py::test_boto_iam - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV4Hand...
FAILED tests/integration/test_requests.py::test_status_code[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceede...
FAILED tests/integration/test_requests.py::test_headers[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded wi...
FAILED tests/integration/test_requests.py::test_body[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded with ...
FAILED tests/integration/test_requests.py::test_get_empty_content_type_json[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max...
FAILED tests/integration/test_requests.py::test_effective_url[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries excee...
FAILED tests/integration/test_requests.py::test_auth[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded with ...
FAILED tests/integration/test_requests.py::test_auth_failed[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceede...
FAILED tests/integration/test_requests.py::test_post[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded with ...
FAILED tests/integration/test_requests.py::test_redirects[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded ...
FAILED tests/integration/test_requests.py::test_gzip[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded with ...
FAILED tests/integration/test_requests.py::test_session_created_before_use_cassette_is_patched[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1...
FAILED tests/integration/test_requests.py::test_nested_cassettes_with_session_created_before_nesting[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127...
FAILED tests/integration/test_requests.py::test_post_file[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded ...
FAILED tests/integration/test_requests.py::test_filter_post_params[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries ...
FAILED tests/integration/test_requests.py::test_post_unicode_match_on_body[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max ...
FAILED tests/integration/test_requests.py::test_cross_scheme - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36041): Max retries exceeded with...
=================================================== 20 failed, 366 passed, 4 skipped, 20 xfailed, 176 warnings in 46.89s ===================================================
```

Dio you have aby idea why those younits may be failing?
And/or is it anything else what I can try to diagnose those fails? 🤔

Summary from testing latest 4.3.0
<details>

```console
===================================================================================== warnings summary ======================================================================================
../../../../../usr/lib/python3.8/site-packages/boto/plugin.py:40
  /usr/lib/python3.8/site-packages/boto/plugin.py:40: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

tests/integration/test_tornado.py:47
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:47: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:59
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:59: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:71
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:71: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:84
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:84: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:97
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:97: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:112
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:112: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:135
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:135: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:150
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:150: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:162
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:162: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:181
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:181: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:206
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:206: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:218
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:218: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:236
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:236: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:252
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:252: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:268
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:268: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:282
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:282: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:289
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:289: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:303
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:303: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:316
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:316: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:331
  /home/tkloczko/rpmbuild/BUILD/vcrpy-4.3.0/tests/integration/test_tornado.py:331: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:47
  tests/integration/test_tornado.py:47: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_status_code will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:59
  tests/integration/test_tornado.py:59: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_headers will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:71
  tests/integration/test_tornado.py:71: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_body will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:84
  tests/integration/test_tornado.py:84: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_effective_url will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:97
  tests/integration/test_tornado.py:97: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:112
  tests/integration/test_tornado.py:112: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:135
  tests/integration/test_tornado.py:135: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_post will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:150
  tests/integration/test_tornado.py:150: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_redirects will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:162
  tests/integration/test_tornado.py:162: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:181
  tests/integration/test_tornado.py:181: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_gzip will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:206
  tests/integration/test_tornado.py:206: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:218
  tests/integration/test_tornado.py:218: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:235
  tests/integration/test_tornado.py:235: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:252
  tests/integration/test_tornado.py:252: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:267
  tests/integration/test_tornado.py:267: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:282
  tests/integration/test_tornado.py:282: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:289
  tests/integration/test_tornado.py:289: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:303
  tests/integration/test_tornado.py:303: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:316
  tests/integration/test_tornado.py:316: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:331
  tests/integration/test_tornado.py:331: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
    @pytest.mark.gen_test

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/integration/test_boto3.py:60: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
SKIPPED [1] tests/integration/test_boto3.py:77: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
SKIPPED [1] tests/integration/test_requests.py:117: condition: sys.version_info >= (3, 6)
XFAIL tests/integration/test_tornado.py::test_status_code - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_status_code will be ignored
XFAIL tests/integration/test_tornado.py::test_headers - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_headers will be ignored
XFAIL tests/integration/test_tornado.py::test_body - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_body will be ignored
XFAIL tests/integration/test_tornado.py::test_effective_url - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_effective_url will be ignored
XFAIL tests/integration/test_tornado.py::test_auth - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth will be ignored
XFAIL tests/integration/test_tornado.py::test_auth_failed - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
XFAIL tests/integration/test_tornado.py::test_post - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_post will be ignored
XFAIL tests/integration/test_tornado.py::test_redirects - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_redirects will be ignored
XFAIL tests/integration/test_tornado.py::test_cross_scheme - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
XFAIL tests/integration/test_tornado.py::test_gzip - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_gzip will be ignored
XFAIL tests/integration/test_tornado.py::test_https_with_cert_validation_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raises_in_future - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raise_error_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raises_in_future - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raise_error_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_with_decorator_use_cassette - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_exception_can_be_caught - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_references_get_patched - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_instances_get_patched - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_request_time_is_set - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
FAILED tests/integration/test_boto.py::test_boto_without_vcr - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
FAILED tests/integration/test_boto.py::test_boto_medium_difficulty - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
FAILED tests/integration/test_boto.py::test_boto_hardcore_mode - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
FAILED tests/integration/test_boto.py::test_boto_iam - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV4Handler'] Check your credentials
FAILED tests/integration/test_requests.py::test_status_code[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VER...
FAILED tests/integration/test_requests.py::test_headers[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VER...
FAILED tests/integration/test_requests.py::test_body[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /bytes/1024 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_get_empty_content_type_json[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /status/200 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_effective_url[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /redirect-to?url=/html (Caused by SSLError(SSLCertVerificationError(1, '...
FAILED tests/integration/test_requests.py::test_auth[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /basic-auth/user/passwd (Caused by SSLError(SSLCertVerificationError(1, ...
FAILED tests/integration/test_requests.py::test_auth_failed[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /basic-auth/user/passwd (Caused by SSLError(SSLCertVerificationError(1, ...
FAILED tests/integration/test_requests.py::test_post[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_redirects[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /redirect-to?url=bytes/1024 (Caused by SSLError(SSLCertVerificationError...
FAILED tests/integration/test_requests.py::test_gzip[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /gzip (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_session_created_before_use_cassette_is_patched[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /bytes/1024 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_nested_cassettes_with_session_created_before_nesting[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /bytes/1024 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_post_file[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_filter_post_params[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_post_unicode_match_on_body[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_cross_scheme - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=45907): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VER...
======================================================= 20 failed, 388 passed, 3 skipped, 20 xfailed, 41 warnings in 74.26s (0:01:14) =======================================================
```
</details>

Just FTR.
Here is summary pytest output on testing 5.0.0:
<details>

```console
===================================================================================== warnings summary ======================================================================================
../../../../../usr/lib/python3.8/site-packages/boto/plugin.py:40
  /usr/lib/python3.8/site-packages/boto/plugin.py:40: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

tests/integration/test_tornado.py:46
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:46: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:58
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:58: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:70
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:70: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:83
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:83: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:96
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:96: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:111
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:111: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:134
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:134: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:149
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:149: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:161
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:161: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:180
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:180: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:205
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:205: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:217
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:217: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:235
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:235: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:251
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:251: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:267
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:267: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:281
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:281: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:288
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:288: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:302
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:302: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:315
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:315: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:330
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:330: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:46
  tests/integration/test_tornado.py:46: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_status_code will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:58
  tests/integration/test_tornado.py:58: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_headers will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:70
  tests/integration/test_tornado.py:70: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_body will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:83
  tests/integration/test_tornado.py:83: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_effective_url will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:96
  tests/integration/test_tornado.py:96: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:111
  tests/integration/test_tornado.py:111: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:134
  tests/integration/test_tornado.py:134: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_post will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:149
  tests/integration/test_tornado.py:149: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_redirects will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:161
  tests/integration/test_tornado.py:161: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:180
  tests/integration/test_tornado.py:180: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_gzip will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:205
  tests/integration/test_tornado.py:205: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:217
  tests/integration/test_tornado.py:217: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:234
  tests/integration/test_tornado.py:234: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:251
  tests/integration/test_tornado.py:251: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:266
  tests/integration/test_tornado.py:266: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:281
  tests/integration/test_tornado.py:281: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:288
  tests/integration/test_tornado.py:288: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:302
  tests/integration/test_tornado.py:302: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:315
  tests/integration/test_tornado.py:315: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:330
  tests/integration/test_tornado.py:330: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
    @pytest.mark.gen_test

tests/integration/test_requests.py::test_effective_url[http]
tests/integration/test_requests.py::test_redirects[http]
  /usr/lib/python3.8/site-packages/werkzeug/datastructures/headers.py:330: DeprecationWarning: Passing bytes as a header value is deprecated and will not be supported in Werkzeug 3.0.
    _value = _str_header_value(_value)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== short test summary info ==================================================================================
SKIPPED [1] tests/integration/test_boto3.py:60: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
SKIPPED [1] tests/integration/test_boto3.py:77: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
XFAIL tests/integration/test_tornado.py::test_status_code - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_status_code will be ignored
XFAIL tests/integration/test_tornado.py::test_headers - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_headers will be ignored
XFAIL tests/integration/test_tornado.py::test_body - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_body will be ignored
XFAIL tests/integration/test_tornado.py::test_effective_url - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_effective_url will be ignored
XFAIL tests/integration/test_tornado.py::test_auth - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth will be ignored
XFAIL tests/integration/test_tornado.py::test_auth_failed - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
XFAIL tests/integration/test_tornado.py::test_post - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_post will be ignored
XFAIL tests/integration/test_tornado.py::test_redirects - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_redirects will be ignored
XFAIL tests/integration/test_tornado.py::test_cross_scheme - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
XFAIL tests/integration/test_tornado.py::test_gzip - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_gzip will be ignored
XFAIL tests/integration/test_tornado.py::test_https_with_cert_validation_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raises_in_future - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raise_error_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raises_in_future - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raise_error_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_with_decorator_use_cassette - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_exception_can_be_caught - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_references_get_patched - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_instances_get_patched - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_request_time_is_set - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
ERROR tests/integration/test_aiohttp.py::test_cookies[http] - KeyError: 'REQUESTS_CA_BUNDLE'
ERROR tests/integration/test_aiohttp.py::test_cookies_redirect[http] - KeyError: 'REQUESTS_CA_BUNDLE'
ERROR tests/integration/test_aiohttp.py::test_cookies[https] - KeyError: 'REQUESTS_CA_BUNDLE'
ERROR tests/integration/test_aiohttp.py::test_cookies_redirect[https] - KeyError: 'REQUESTS_CA_BUNDLE'
FAILED tests/integration/test_urllib2.py::test_random_body[http] - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED tests/integration/test_urllib2.py::test_multiple_requests[http] - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED tests/integration/test_requests.py::test_status_code[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VER...
FAILED tests/integration/test_requests.py::test_headers[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VER...
FAILED tests/integration/test_requests.py::test_body[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /bytes/1024 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_get_empty_content_type_json[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /status/200 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_effective_url[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /redirect-to?url=/html (Caused by SSLError(SSLCertVerificationError(1, '...
FAILED tests/integration/test_requests.py::test_auth[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /basic-auth/user/passwd (Caused by SSLError(SSLCertVerificationError(1, ...
FAILED tests/integration/test_requests.py::test_auth_failed[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /basic-auth/user/passwd (Caused by SSLError(SSLCertVerificationError(1, ...
FAILED tests/integration/test_requests.py::test_post[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_redirects[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /redirect-to?url=bytes/1024 (Caused by SSLError(SSLCertVerificationError...
FAILED tests/integration/test_requests.py::test_gzip__decode_compressed_response_false[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /gzip (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_gzip__decode_compressed_response_true[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /gzip (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_session_created_before_use_cassette_is_patched[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /bytes/1024 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_nested_cassettes_with_session_created_before_nesting[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /bytes/1024 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTI...
FAILED tests/integration/test_requests.py::test_post_file[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_filter_post_params[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_requests.py::test_post_unicode_match_on_body[https] - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: /post (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE...
FAILED tests/integration/test_urllib2.py::test_random_body[https] - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED tests/integration/test_urllib2.py::test_multiple_requests[https] - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED tests/integration/test_boto.py::test_boto_without_vcr - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
FAILED tests/integration/test_boto.py::test_boto_medium_difficulty - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
FAILED tests/integration/test_boto.py::test_boto_hardcore_mode - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV1Handler'] Check your credentials
FAILED tests/integration/test_boto.py::test_boto_iam - boto.exception.NoAuthHandlerFound: No handler was ready to authenticate. 1 handlers were checked. ['HmacAuthV4Handler'] Check your credentials
FAILED tests/integration/test_record_mode.py::test_new_episodes_record_mode_two_times - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED tests/integration/test_requests.py::test_cross_scheme - requests.exceptions.SSLError: HTTPSConnectionPool(host='127.0.0.1', port=36847): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VER...
FAILED tests/unit/test_unittest.py::test_get_vcr_with_matcher - assert 0 == 1
FAILED tests/unit/test_unittest.py::test_testcase_playback - AttributeError: 'MyTest' object has no attribute 'response'
======================================================= 28 failed, 394 passed, 2 skipped, 20 xfailed, 43 warnings, 4 errors in 57.66s =======================================================
```
</details>

Full log is in attachment [python-vcrpy-pytest.txt](https://github.com/kevin1024/vcrpy/files/11948643/python-vcrpy-pytest.txt)

Here is list of installed modules in build env
<details>

```console
Package                       Version
----------------------------- ---------
aiohttp                       3.8.4
aiosignal                     1.3.1
alabaster                     0.7.13
anyio                         3.6.2
asttokens                     2.2.1
async-timeout                 4.0.2
attrs                         23.1.0
Babel                         2.12.1
backcall                      0.2.0
blinker                       1.6.2
boto                          2.49.0
boto3                         1.26.164
botocore                      1.29.164
Brotli                        1.0.9
build                         0.10.0
certifi                       2022.12.7
charset-normalizer            3.1.0
click                         8.1.3
decorator                     5.1.1
distro                        1.8.0
docutils                      0.19
exceptiongroup                1.1.1
executing                     1.2.0
Flask                         2.2.5
frozenlist                    1.3.3
gpg                           1.20.0
h11                           0.14.0
httpbin                       0.7.0
httpcore                      0.17.2
httplib2                      0.21.0
httpx                         0.24.1
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.7.0
iniconfig                     2.0.0
installer                     0.7.0
ipython                       8.12.0
itsdangerous                  2.1.2
jedi                          0.18.2
Jinja2                        3.1.2
jmespath                      1.0.1
libcomps                      0.1.19
MarkupSafe                    2.1.2
matplotlib-inline             0.1.6
multidict                     6.0.4
packaging                     23.1
parso                         0.8.3
pexpect                       4.8.0
pickleshare                   0.7.5
pluggy                        1.0.0
prompt-toolkit                3.0.38
ptyprocess                    0.7.0
pure-eval                     0.2.2
Pygments                      2.15.1
pyparsing                     3.1.0
pyproject_hooks               1.0.0
pytest                        7.4.0
pytest-aiohttp                1.0.4
pytest-asyncio                0.21.0
pytest-httpbin                2.0.0
pytest-mock                   3.11.1
python-dateutil               2.8.2
pytz                          2023.2
PyYAML                        6.0
raven                         6.10.0
requests                      2.31.0
s3transfer                    0.6.1
setuptools                    68.0.0
six                           1.16.0
sniffio                       1.3.0
snowballstemmer               2.2.0
Sphinx                        6.2.1
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
stack-data                    0.6.2
tomli                         2.0.1
tornado                       6.3.2
traitlets                     5.9.0
typing_extensions             4.7.1
urllib3                       1.26.15
wcwidth                       0.2.6
Werkzeug                      2.3.6
wheel                         0.40.0
wrapt                         1.14.1
yarl                          1.9.2
zipp                          3.15.0
```
</details>

> @kloczek you were asking for `REQUESTS_CA_BUNDLE`. The answer is:
> 
> ```
> REQUESTS_CA_BUNDLE=`python -m pytest_httpbin.certs`
> ```

```
[tkloczko@pers-jacek SPECS]$ python3 -m pytest_httpbin.certs
/usr/lib/python3.8/site-packages/pytest_httpbin/certs/cacert.pem
```

Hmm I must patch that module to use cert file from /ets/pki.
After pass correct file
<details>

```console
+ export REQUESTS_CA_BUNDLE=/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-vcrpy-5.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-vcrpy-5.0.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
============================= test session starts ==============================
platform linux -- Python 3.8.17, pytest-7.4.0, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0
configfile: pyproject.toml
plugins: anyio-3.6.2, asyncio-0.21.0, aiohttp-1.0.4, httpbin-2.0.0, mock-3.11.1
asyncio: mode=strict
collected 448 items

tests/integration/test_aiohttp.py ..................................EE   [  8%]
tests/integration/test_httplib2.py ........                              [  9%]
tests/integration/test_requests.py ................                      [ 13%]
tests/integration/test_urllib2.py .F.F....                               [ 15%]
tests/integration/test_urllib3.py .......                                [ 16%]
tests/integration/test_aiohttp.py EE                                     [ 17%]
tests/integration/test_httplib2.py ........                              [ 18%]
tests/integration/test_requests.py pytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
Fpytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
F                      [ 22%]
tests/integration/test_urllib2.py .F.F....                               [ 24%]
tests/integration/test_urllib3.py .......                                [ 25%]
tests/integration/test_aiohttp.py ..                                     [ 26%]
tests/integration/test_basic.py .....                                    [ 27%]
tests/integration/test_boto.py .FFFF                                     [ 28%]
tests/integration/test_boto3.py ss                                       [ 29%]
tests/integration/test_config.py ...........                             [ 31%]
tests/integration/test_disksaver.py ....                                 [ 32%]
tests/integration/test_filter.py ..........                              [ 34%]
tests/integration/test_httplib2.py ..                                    [ 35%]
tests/integration/test_httpx.py ........................................ [ 43%]
........                                                                 [ 45%]
tests/integration/test_ignore.py ....                                    [ 46%]
tests/integration/test_matchers.py ..............                        [ 49%]
tests/integration/test_multiple.py .                                     [ 50%]
tests/integration/test_proxy.py .                                        [ 50%]
tests/integration/test_record_mode.py ....F...                           [ 52%]
tests/integration/test_register_matcher.py ....                          [ 52%]
tests/integration/test_register_persister.py ...                         [ 53%]
tests/integration/test_register_serializer.py .                          [ 53%]
tests/integration/test_request.py ..                                     [ 54%]
tests/integration/test_requests.py ..pytest-httpbin server hit an exception serving request: [SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:1131)
attempting to ignore so the rest of the tests can run
F...                                [ 55%]
tests/integration/test_stubs.py ....                                     [ 56%]
tests/integration/test_tornado.py xxxxxxxxxxxxxxxxxxxx                   [ 60%]
tests/integration/test_urllib2.py ..                                     [ 61%]
tests/integration/test_urllib3.py ....                                   [ 62%]
tests/integration/test_wild.py ......                                    [ 63%]
tests/unit/test_cassettes.py ...............................             [ 70%]
tests/unit/test_errors.py ....                                           [ 71%]
tests/unit/test_filters.py ........................                      [ 76%]
tests/unit/test_json_serializer.py .                                     [ 77%]
tests/unit/test_matchers.py .........................                    [ 82%]
tests/unit/test_migration.py ...                                         [ 83%]
tests/unit/test_persist.py ....                                          [ 84%]
tests/unit/test_request.py .................                             [ 87%]
tests/unit/test_response.py ....                                         [ 88%]
tests/unit/test_serialize.py ...............                             [ 92%]
tests/unit/test_stubs.py ..                                              [ 92%]
tests/unit/test_unittest.py .......FF                                    [ 94%]
tests/unit/test_vcr.py .......................                           [ 99%]
tests/unit/test_vcr_import.py .                                          [100%]

==================================== ERRORS ====================================

[...]

=============================== warnings summary ===============================
../../../../../usr/lib/python3.8/site-packages/boto/plugin.py:40
  /usr/lib/python3.8/site-packages/boto/plugin.py:40: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

tests/integration/test_tornado.py:46
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:46: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:58
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:58: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:70
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:70: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:83
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:83: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:96
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:96: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:111
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:111: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:134
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:134: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:149
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:149: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:161
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:161: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:180
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:180: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:205
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:205: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:217
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:217: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:235
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:235: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:251
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:251: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:267
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:267: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:281
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:281: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:288
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:288: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:302
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:302: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:315
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:315: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:330
  /home/tkloczko/rpmbuild/BUILD/vcrpy-5.0.0/tests/integration/test_tornado.py:330: PytestUnknownMarkWarning: Unknown pytest.mark.gen_test - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.gen_test

tests/integration/test_tornado.py:46
  tests/integration/test_tornado.py:46: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_status_code will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:58
  tests/integration/test_tornado.py:58: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_headers will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:70
  tests/integration/test_tornado.py:70: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_body will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:83
  tests/integration/test_tornado.py:83: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_effective_url will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:96
  tests/integration/test_tornado.py:96: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:111
  tests/integration/test_tornado.py:111: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:134
  tests/integration/test_tornado.py:134: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_post will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:149
  tests/integration/test_tornado.py:149: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_redirects will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:161
  tests/integration/test_tornado.py:161: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:180
  tests/integration/test_tornado.py:180: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_gzip will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:205
  tests/integration/test_tornado.py:205: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:217
  tests/integration/test_tornado.py:217: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:234
  tests/integration/test_tornado.py:234: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:251
  tests/integration/test_tornado.py:251: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:266
  tests/integration/test_tornado.py:266: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
    @pytest.mark.skipif(not supports_raise_error, reason="raise_error unavailable in tornado <= 3")

tests/integration/test_tornado.py:281
  tests/integration/test_tornado.py:281: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:288
  tests/integration/test_tornado.py:288: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:302
  tests/integration/test_tornado.py:302: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:315
  tests/integration/test_tornado.py:315: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
    @pytest.mark.gen_test

tests/integration/test_tornado.py:330
  tests/integration/test_tornado.py:330: PytestCollectionWarning: yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
    @pytest.mark.gen_test

tests/integration/test_requests.py::test_effective_url[http]
tests/integration/test_requests.py::test_redirects[http]
  /usr/lib/python3.8/site-packages/werkzeug/datastructures/headers.py:330: DeprecationWarning: Passing bytes as a header value is deprecated and will not be supported in Werkzeug 3.0.
    _value = _str_header_value(_value)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
SKIPPED [1] tests/integration/test_boto3.py:60: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
SKIPPED [1] tests/integration/test_boto3.py:77: Encrypted Environment Variables from Travis Repository Settings are disabled on PRs from forks. https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
XFAIL tests/integration/test_tornado.py::test_status_code - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_status_code will be ignored
XFAIL tests/integration/test_tornado.py::test_headers - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_headers will be ignored
XFAIL tests/integration/test_tornado.py::test_body - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_body will be ignored
XFAIL tests/integration/test_tornado.py::test_effective_url - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_effective_url will be ignored
XFAIL tests/integration/test_tornado.py::test_auth - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth will be ignored
XFAIL tests/integration/test_tornado.py::test_auth_failed - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_auth_failed will be ignored
XFAIL tests/integration/test_tornado.py::test_post - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_post will be ignored
XFAIL tests/integration/test_tornado.py::test_redirects - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_redirects will be ignored
XFAIL tests/integration/test_tornado.py::test_cross_scheme - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cross_scheme will be ignored
XFAIL tests/integration/test_tornado.py::test_gzip - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_gzip will be ignored
XFAIL tests/integration/test_tornado.py::test_https_with_cert_validation_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_https_with_cert_validation_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raises_in_future - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_unsupported_features_raise_error_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_unsupported_features_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raises_in_future - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raises_in_future will be ignored
XFAIL tests/integration/test_tornado.py::test_cannot_overwrite_cassette_raise_error_disabled - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_cannot_overwrite_cassette_raise_error_disabled will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_with_decorator_use_cassette - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_with_decorator_use_cassette will be ignored
XFAIL tests/integration/test_tornado.py::test_tornado_exception_can_be_caught - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_tornado_exception_can_be_caught will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_references_get_patched - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_references_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_existing_instances_get_patched - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_existing_instances_get_patched will be ignored
XFAIL tests/integration/test_tornado.py::test_request_time_is_set - reason: [NOTRUN] yield tests were removed in pytest 4.0 - test_request_time_is_set will be ignored
ERROR tests/integration/test_aiohttp.py::test_cookies[http] - ssl.SSLError: [...
ERROR tests/integration/test_aiohttp.py::test_cookies_redirect[http] - ssl.SS...
ERROR tests/integration/test_aiohttp.py::test_cookies[https] - ssl.SSLError: ...
ERROR tests/integration/test_aiohttp.py::test_cookies_redirect[https] - ssl.S...
FAILED tests/integration/test_urllib2.py::test_random_body[http] - urllib.err...
FAILED tests/integration/test_urllib2.py::test_multiple_requests[http] - urll...
FAILED tests/integration/test_requests.py::test_status_code[https] - requests...
FAILED tests/integration/test_requests.py::test_headers[https] - requests.exc...
FAILED tests/integration/test_requests.py::test_body[https] - requests.except...
FAILED tests/integration/test_requests.py::test_get_empty_content_type_json[https]
FAILED tests/integration/test_requests.py::test_effective_url[https] - reques...
FAILED tests/integration/test_requests.py::test_auth[https] - requests.except...
FAILED tests/integration/test_requests.py::test_auth_failed[https] - requests...
FAILED tests/integration/test_requests.py::test_post[https] - requests.except...
FAILED tests/integration/test_requests.py::test_redirects[https] - requests.e...
FAILED tests/integration/test_requests.py::test_gzip__decode_compressed_response_false[https]
FAILED tests/integration/test_requests.py::test_gzip__decode_compressed_response_true[https]
FAILED tests/integration/test_requests.py::test_session_created_before_use_cassette_is_patched[https]
FAILED tests/integration/test_requests.py::test_nested_cassettes_with_session_created_before_nesting[https]
FAILED tests/integration/test_requests.py::test_post_file[https] - requests.e...
FAILED tests/integration/test_requests.py::test_filter_post_params[https] - r...
FAILED tests/integration/test_requests.py::test_post_unicode_match_on_body[https]
FAILED tests/integration/test_urllib2.py::test_random_body[https] - urllib.er...
FAILED tests/integration/test_urllib2.py::test_multiple_requests[https] - url...
FAILED tests/integration/test_boto.py::test_boto_without_vcr - boto.exception...
FAILED tests/integration/test_boto.py::test_boto_medium_difficulty - boto.exc...
FAILED tests/integration/test_boto.py::test_boto_hardcore_mode - boto.excepti...
FAILED tests/integration/test_boto.py::test_boto_iam - boto.exception.NoAuthH...
FAILED tests/integration/test_record_mode.py::test_new_episodes_record_mode_two_times
FAILED tests/integration/test_requests.py::test_cross_scheme - requests.excep...
FAILED tests/unit/test_unittest.py::test_get_vcr_with_matcher - assert 0 == 1
FAILED tests/unit/test_unittest.py::test_testcase_playback - AttributeError: ...
= 28 failed, 394 passed, 2 skipped, 20 xfailed, 43 warnings, 4 errors in 188.28s (0:03:08) =
```
</details>

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,7 +37,7 @@ tests = [
     "aiohttp",
     "boto3",
     "cryptography",
-    "httpbin",
+    "httpbin>=0.10.3",
     "httplib2",
     "httpx",
     "httpx-curl-cffi",
@@ -52,9 +52,9 @@ tests = [
     "requests>=2.22.0",
     "tornado",
     "urllib3",
-    "werkzeug==2.0.3",
 ]
 tests-niquests = [
+    "httpbin>=0.10.3",
     "pytest",
     "pytest-aiohttp",
     "pytest-asyncio",

--- a/tests/integration/test_stale_connection.py
+++ b/tests/integration/test_stale_connection.py
@@ -1,0 +1,59 @@
+"""Regression test for reuse of a stale (server-closed) keep-alive connection.
+
+vcrpy replaces urllib3's connection classes with socketless stubs and used to
+force ``is_connection_dropped`` to always return ``False`` (a Windows playback
+fix, #116). Applied to *real* recording connections, that disabled urllib3's
+stale-connection check, so a keep-alive connection the server had already closed
+would be reused -- raising ``ProtocolError('Connection aborted', ...)`` (an
+intermittent BrokenPipe/RemoteDisconnected in CI).
+"""
+
+import socket
+import threading
+import time
+
+import urllib3
+
+import vcr
+
+
+def _keepalive_then_close_server():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    sock.bind(("127.0.0.1", 0))
+    sock.listen(5)
+    port = sock.getsockname()[1]
+
+    def serve():
+        while True:
+            try:
+                conn, _ = sock.accept()
+            except OSError:
+                return
+            try:
+                conn.recv(65536)
+                conn.sendall(
+                    b"HTTP/1.1 200 OK\r\nContent-Length: 2\r\nConnection: keep-alive\r\n\r\nok",
+                )
+                # Hand the keep-alive connection back, then drop it server-side,
+                # leaving the client with a pooled-but-dead socket.
+                time.sleep(0.02)
+                conn.close()
+            except OSError:
+                pass
+
+    threading.Thread(target=serve, daemon=True).start()
+    return sock, port
+
+
+def test_stale_pooled_connection_is_not_reused(tmpdir):
+    server_sock, port = _keepalive_then_close_server()
+    try:
+        pool = urllib3.HTTPConnectionPool("127.0.0.1", port, maxsize=1)
+        with vcr.use_cassette(str(tmpdir.join("stale.yml"))):
+            pool.urlopen("GET", "/", retries=False, preload_content=True)
+            time.sleep(0.25)  # let the server's close land on the pooled socket
+            response = pool.urlopen("GET", "/", retries=False, preload_content=True)
+            assert response.status == 200
+    finally:
+        server_sock.close()

--- a/vcr/patch.py
+++ b/vcr/patch.py
@@ -480,11 +480,30 @@ class CassettePatcherBuilder:
         https_connection_remover = ConnectionRemover(
             self._get_cassette_subclass(stubs.VCRRequestsHTTPSConnection),
         )
+
+        real_is_connection_dropped = cpool.is_connection_dropped
+
+        def is_connection_dropped(connection):
+            # vcrpy swaps urllib3's connection classes for socketless stubs.
+            # During playback a stub has no live socket, so urllib3's normal
+            # check would treat it as dropped and discard it -- originally a
+            # Windows playback fix (#116). But a stub that is *recording* wraps a
+            # real connection, and we must still run urllib3's real check against
+            # that underlying connection. Otherwise urllib3 reuses a keep-alive
+            # connection the server has already closed and raises a "Connection
+            # aborted" ProtocolError (the intermittent BrokenPipe in CI).
+            real_connection = getattr(connection, "real_connection", None)
+            if real_connection is None:
+                return real_is_connection_dropped(connection)
+            if getattr(real_connection, "sock", None) is None:
+                return False
+            return real_is_connection_dropped(real_connection)
+
         mock_triples = (
             (conn, "VerifiedHTTPSConnection", stubs.VCRRequestsHTTPSConnection),
             (conn, "HTTPConnection", stubs.VCRRequestsHTTPConnection),
             (conn, "HTTPSConnection", stubs.VCRRequestsHTTPSConnection),
-            (cpool, "is_connection_dropped", mock.Mock(return_value=False)),  # Needed on Windows only
+            (cpool, "is_connection_dropped", is_connection_dropped),
             (cpool.HTTPConnectionPool, "ConnectionCls", stubs.VCRRequestsHTTPConnection),
             (cpool.HTTPSConnectionPool, "ConnectionCls", stubs.VCRRequestsHTTPSConnection),
         )

